icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -4313,11 +4313,9 @@ SimpleDateFormat::tzFormat(UErrorCode &status) const {
         {
```

Comment:
The double-checked lock idiom used here is also unsafe. If could be fixed by making fTimeZoneFormat an atomic type, and using the appropriate atomic access functions, but doing so would be awkward and complicated because fTimeZoneFormat is declared in a public header. The most expedient fix is probably to ditch the double checked idiom, and just use the mutex.